tests/templates.py

```diff
@@ -79,20 +95,31 @@ def render_script(template: str, out_path: str, params: Dict):
 
 
 def autogen_cargo(conf_file, yaml: Dict):
-    cargo = yaml.get("cargo")
-    if cargo and isinstance(cargo, Dict):
-        ag = cargo.get("autogen")
-        if ag and isinstance(ag, bool):
-            params = {}
-            rustflags = cargo.get("rustflags")
-            if rustflags and isinstance(rustflags, str):
-                params['extra_rustflags'] = rustflags
-
-            out_path = os.path.join(
-                os.path.dirname(conf_file),
-                "cargo.gen.sh"
-            )
-            render_script(CARGO_SH, out_path, params)
+    def render_stage(stage_conf: Dict, filename: str) -> bool:
+        if not stage_conf or not isinstance(stage_conf, Dict):
```

Comment:
`Dict` is the `typing` type, not the actual `dict` class, so I'm not sure this works? 

Also, if the `not stage_conf` is to test for `None`, then it's clearer to write `stage_conf is not None` and have `stage_conf: Dict | None`.  If it's to also test for an empty `dict`, then it makes more sense to come after the `isinstance`.

Reply:
let me know if what I did isn't what you're thinking